src/Catalog/ReadOnlyGraph.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using VDS.RDF;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    /// <summary>
+    /// An immutable graph supporting multi-threaded read-only access. According to dotNetRDF documentation, the
+    /// <see cref="Graph"/> class is thread-safe when used in a read-only manner.
+    /// <see cref="http://www.dotnetrdf.org/api/html/T_VDS_RDF_Graph.htm"/>
+    /// </summary>
+    public class ReadOnlyGraph : Graph
+    {
+        private readonly bool _isReadOnly;
```

Comment:
Oh I see--the `Graph` constructor must call these methods internally, so if we override them then it fails initialization. Interesting...